compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -2129,7 +2129,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             let target_ty = self
                 .autoderef(sugg_span, rcvr_ty)
                 .find(|(rcvr_ty, _)| {
-                    DeepRejectCtxt { treat_obligation_params: TreatParams::AsCandidateKey }
+                    DeepRejectCtxt::new(self.tcx, TreatParams::ForLookup)
```

Comment:
I'm pretty sure this being `AsCandidateKey` was wrong, since we're trying to match an impl.

Reply:
`AsCandidateKey` is strictly weaker than `ForLookup` as it's used when we'd later replace all params with infer vars.

Using `ForLookup` is correct here :+1: